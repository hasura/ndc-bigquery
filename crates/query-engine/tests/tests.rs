mod common;

#[test]
fn it_converts_select_with_limit() {
    let result = common::test_translation("select_with_limit").unwrap();
    insta::assert_snapshot!(result);
}

#[test]
fn it_select_where_string() {
    let result = common::test_translation("select_where_string").unwrap();
    insta::assert_snapshot!(result);
}

#[test]
fn it_aggregate_count_albums() {
    let result = common::test_translation("aggregate_count_albums").unwrap();
    insta::assert_snapshot!(result);
}

#[test]
fn it_aggregate_distinct_albums() {
    let result = common::test_translation("aggregate_distinct_albums").unwrap();
    insta::assert_snapshot!(result);
}

#[test]
fn it_aggregate_function_albums() {
    let result = common::test_translation("aggregate_function_albums").unwrap();
    insta::assert_snapshot!(result);
}
