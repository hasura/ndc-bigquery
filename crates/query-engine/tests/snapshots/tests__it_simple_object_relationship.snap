---
source: crates/query-engine/tests/tests.rs
expression: result
---
SELECT row_to_json("universe") AS "universe" FROM (SELECT * FROM (SELECT coalesce(json_agg(row_to_json("rows")), '[]') AS "rows" FROM (SELECT "Artist"."Artist" AS "Artist" FROM "public"."Album" AS "Album" LEFT OUTER JOIN LATERAL (SELECT json_build_object('rows', coalesce(json_agg(row_to_json("Artist")), '[]')) AS "Artist" FROM (SELECT "Artist"."Name" AS "Name" FROM "public"."Artist" AS "Artist" WHERE (true AND ("Album"."ArtistId" = "Artist"."ArtistId"))) AS "Artist") AS "Artist" ON ('true')  LIMIT 5) AS "rows") AS "rows") AS "universe"

[]
